### ISSUE TITLE
horovod ingore pytorch optimizer l2 regularization function

### ISSUE THREAD

**Issue by jasstionzyf** [horovod/horovod#2701]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.7.1
3. Horovod version:  0.21.3
4. MPI version:
5. CUDA version: 10.1
6. NCCL version:
7. Python version: 3.7
8. Spark / PySpark version:
9. OS and version:
10. GCC version:
11. CMake version:

**Bug report:**
 ```

optim.SGD(modelParams, lr=lr, momentum=0.9,weight_decay=weight_decay)
optimizer = DistributedOptimizer(optimizer, named_parameters=model.named_parameters(),
                                                    compression=compression)  


lossValue.backward()

optimizer.step()



```

above code will not apply l2 regularization  within SGD,
as a result, l2 loss will keep growing
except you add l2 loss to lossValue
`(lossValue+regLossValue).backward()`

**Comment by tgaddair** [comment]
Thanks for reporting @jasstionzyf. To help with debugging this, can you provide a complete repro script we can run, along with describing the expected behavior?
